internal/gitlab/codequality_test.go

```diff
@@ -93,3 +95,186 @@ func TestCodeQuality(t *testing.T) {
 		})
 	}
 }
+
+func TestCodeQuality_DetectChangedLine(t *testing.T) {
```

Comment:
This should be a table test with more test cases. The `testdata` directory should have multiple different examples of changed lines. Some edge cases I would like to see:
* No lines changed
* All lines changed
* One line changed
* Only final line changed
* Only change is one final line added